### ISSUE TITLE
add layer_norm does not work (optimization)

### ISSUE THREAD

**Issue by louieworth** [google-research/jaxpruner#7]
I am trying to add `layer_normalization` trick to stabilize the training process. However, I found it did not train (the total performance does not increase, without learning signals) after the operation of `pruner.wrap_optax()`. Do you have any suggestions about how to fix it?

here is some source code:
```
class MLP(nn.Module):
    hidden_dims: Sequence[int] # (512, 512)
    activations: Callable[[jnp.ndarray], jnp.ndarray] = nn.relu
    activate_final: int = False
    layer_normalization: bool = False
    dropout_rate: Optional[float] = None
    
    @nn.compact
    def __call__(self, x: jnp.ndarray, training: bool = False) -> jnp.ndarray:
        for i, size in enumerate(self.hidden_dims):
            x = nn.Dense(size, kernel_init=default_init())(x)
            if self.layer_normalization:
                x = nn.LayerNorm()(x) # add layer normalization 
            if i + 1 < len(self.hidden_dims) or self.activate_final:
                x = self.activations(x)
                if self.dropout_rate is not None:
                    x = nn.Dropout(rate=self.dropout_rate)(
                        x, deterministic=not training)
        return x
```
Here is the performance (Fig. 1)obtained with `nn.LayerNorm()`. Through controlled experiments, the algorithm can achieve high performance without `nn.LayerNorm()`.  The source code is from [jaxrl]()
![layer_norm](https://github.com/google-research/jaxpruner/assets/37494792/aed693b5-a77a-4078-bc84-188c1c10d3f8)
Fig. 1: With layer normalization, there is no learning signal and no performance improvement. 
![no_layer_norm](https://github.com/google-research/jaxpruner/assets/37494792/6a02b54d-e622-4266-a84b-57a8ceeba86b)
Fig. 2: Without layer normalization, the performance of the model significantly increases and then deteriorates during subsequent training. This decline in performance can be attributed to the high sparsity level, approximately 95%.

**Comment by stepp1** [comment]
My first guess would be to try with higher values of the `eps` arg of LayerNorm.

**Comment by louieworth** [comment]
Do you mean the solution to this problem? Could you please more details with it?

> My first guess would be to try with higher values of the `eps` arg of LayerNorm.
